src/ips/graph_utils.rs

```diff
@@ -90,8 +107,9 @@ pub fn construct_graph(nodes: &[Node]) -> Graph<SocketAddr> {
 /// Removes node from the state and updates all indices in the peerlist
 pub fn remove_node(nodes: &mut Vec<Node>, node_idx: usize) {
     let node = nodes[node_idx].clone();
-    for peer_idx in node.connections.iter() {
-        nodes[*peer_idx].connections.retain(|x| *x != node_idx);
+
+    for rnode in nodes.iter_mut() {
+        rnode.connections.retain(|x| *x != node_idx);
```

Comment:
would a combination of `position` and then `remove` be [faster](https://stackoverflow.com/a/44012406)?
```rust
// Ignore if no such element is found
if let Some(pos) = vec.iter().position(|x| *x == needle) {
    vec.remove(pos);
}
```

I tried to imagine a case where we have 1000 nodes, and we need to remove the first node - the `retain` solution would still go through all 1000 nodes, although the `node_idx==0`

Reply:
Changed.